### PR TITLE
feat($goQuery): Experimental integration of Query 

### PR DESCRIPTION
**Pull request by mattcreager**
$goQuery is a service built on `key.query()`.  It accepts a key, room,
filter and options.  It returns a QueryModel. The Model has the standard
`$sync` method which keeps the models value current with that of the
query result set.

``` js
angular.module('ChatApp').controller('ChatCtrl', ['$scope', '$goQuery', function($scope, $query) {
  var filter = { sender: { '$eq': 'GrandMasterLivingston' }};
  var sort = { $name: 'desc' };
  var limit = 20;
  var options = { sort: sort, limit: limit };

  $scope.chat = $query('messages', filter, options);
  $scope.chat.$sync();
}]);
```

Related: #48

### CODE REVIEW

File: lib/util/args.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
+/**
```
Comment (ianlivingstone) [inline lib/util/args.js:1]: cant you use bower to install this? why does it need to be a checked-in dependency

Comment (mattcreager) [inline lib/util/args.js:1]: I'd love to check this in with bower, but we're using component, which this library doesn't offer... I'd fork & and add it, but we're planning to over-haul the build process anyway!

We could also just manage the argument validation, but I like how this library is documented, implemented, and I think it's one positive step towards a cleaner interface.

File: docs/v1/query.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- ###### **$goQuery(keyName, expr, optionsObject)**
+- ###### **$goQuery(keyName, roomName, expr, optionsObject)**
+
+
+## Parameters
+
+| keyName |
+|:---|
+| Type: [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String). |
+| The name of the key to be referenced. [Naming restrictions apply.](Naming restrictions apply) A leading forward-slash ('/') is implied, (e.g. 'foo/bar' is the same as '/foo/bar'). |
```
Comment (colinmacdonald) [inline docs/v1/query.md]: Replace "Naming restrictions apply" with relative link to docs.

File: tests/query_sync.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    describe('$$handleUpdate', function() {
+      var keyUpdateData = [
+        {
+          desc: 'updates the model with object',
+          value: [ { foo: 'bar' } ],
+          context: { position: { current: 0 } },
+          expect: { foo: 'bar' }
+        },
+        // {
```
Comment (colinmacdonald) [inline tests/query_sync.js]: don't forget to uncomment these tests.

File: lib/query_sync.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+'use strict';
+
+var _ = require('lodash');
+
+module.exports = function querySync($parse, $timeout) {
+
+  /**
+   * @public
+   * @param {Object} key - GoInstant key
```
Comment (colinmacdonald) [inline lib/query_sync.js]: Inline docs - should be a "query" object, not a key?

**Comment by colinmacdonald** [comment]
LGTM when :cactus:

**Pull request merged**